### PR TITLE
[ADD] Manufacturing: Allocation reports

### PR DESCRIPTION
**Pull request by hojo-odoo**
This PR adds a new doc detailing how to enable and use allocation reports in the Manufacturing app.

### CODE REVIEW

**Comment by hojo-odoo** [comment]
Hey @larm-odoo,

This PR is ready for review.

Thanks,
John

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [reply content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:33]: Looks like when you first make a product, 'Can Be Sold' is automatically enabled. Not sure if you want to add that as a note, that this is a default? Or link to an 'add a product' doc? Just suggestions.

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:43]:
```suggestion
To allocate products or components from a |MO| to an |SO| or different
|MO|, begin by navigating to :menuselection:`Manufacturing --> Orders --> Manufacturing Orders`.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:47]:
```suggestion

Click :guilabel:`New` to create a new |MO|, and select a product in the :guilabel:`Product` field. Next, specify the quantity to be
produced in the :guilabel:`Quantity` field. Finally, click :guilabel:`Confirm` to confirm the |MO|.
```
I just wanted to have the 'New' part and the edits together in a paragraph- optional suggestion!

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:51]:
```suggestion
not already been allocated.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:54]:
```suggestion
of the product on-hand to fulfill those orders, then an :icon:`fa-list` :guilabel:`Allocation` smart
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:67]:
```suggestion
   If there are **not** any existing |SOs| and |MOs| that require the product, the :icon:`fa-list`
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:53]:
```suggestion
If there **are** existing |SOs| and |MOs| that require the product, **and** there are too few units
```
Typically when I have these words, they are edited by others to be bold but not be in all caps. Just a suggestion to keep docs somewhat similar, but this is optional.

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:57]: same note as above

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:71]:
```suggestion
Click the :icon:`fa-list` :guilabel:`Allocation` smart button  and the guilabel:`MRP Reception Report` for
the |MO| loads.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:73]:
```suggestion
If a finished product is manufactured, this report lists any open customer delivery orders for which
```
not sure if it's a report but it says it's a report in the name, so this is just a suggestion.

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:77]:
```suggestion
If a component is manufactured, this page lists any open |MOs| for which quantities of
the component have yet to be reserved.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:80]: This example could be slightly re-written to be inside an example admonition block. I fin d these super helpful and they stand out to the reader. Just a suggestion!

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:87]: I found this section a little confusing to me. I think having an example admonition where there is a clear example might be helpful. I know it would be for me =)

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:89]: Is it assigned to a quantity? Or a MO? This just sounds odd to me.

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:91]: Quantities is throwing me here, too. They are being assigned to MO's, right? And you are selecting how many of X are being reserved for MO's? I would replace the word quantities to MO's if that is correct.

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (larm-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:102]: This image is confusing for me - I couldn't get my runbots to do this flow, so I couldn't reproduce this image. It's repeating all the same info on each slip- is there a larger pic with more info that you can add here?

**Review by larm-odoo — APPROVED** [review]
Hi @hojo-odoo - all set! I just have a few comments and suggestions, but approving now!

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (hojo-odoo) [reply content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:102]: Each label should have the same info since all of the products have been allocated to the same order.

**Comment by hojo-odoo** [comment]
Hey @ksc-odoo,

This doc is ready for final review.

Thanks,
John

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:17]:
```suggestion
orders |SOs|, or components to specific |MOs|. This ensures the products or components are
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:23]:
```suggestion
To use allocation reports, the *Allocation Report for Manufacturing Orders* feature **must** be enabled.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:25]:
```suggestion
the checkbox next to :guilabel:`Allocation Report for Manufacturing Orders`. Then, click :guilabel:`Save`.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:31]:
```suggestion
For products that are sold, it is also necessary to configure them, so they can be included in |SOs|.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:32]:
```suggestion
To do so, begin by navigating to :menuselection:`Inventory app --> Products --> Products`, and select a
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:33]:
```suggestion
product. Under the :guilabel:`Product Name` on the product form, make sure that the :guilabel:`Can Be Sold` checkbox is ticked.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:42]:
```suggestion
To allocate products or components from an |MO| to an |SO|, or to a different |MO|, begin by navigating to
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:43]:
```suggestion
:menuselection:`Manufacturing app --> Operations --> Manufacturing Orders`. Click :guilabel:`New` to create
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:46]:
```suggestion
On the |MO| form, select a product in the :guilabel:`Product` field, and specify the quantity to be
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [reply content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:32]: Could this be done directly through the Manufacturing app? Could be a nice way to save some time, and make everything complete in one app -- but that's just a suggestion. Totally your call.

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:70]:
```suggestion
Click the :icon:`fa-list` :guilabel:`Allocation` smart button to open the :guilabel:`MRP Reception Report` for
```
OPTIONAL
I fully get the use of italics here because, at this point, the title of the page *technically* isn't being seen. However, I am merely suggesting the use of a GUI here because, if the reader were to follow along as each step is explained, they would have, at this point, clicked the 'Allocation' smart button, which *does* reveal this title on the page. 

That said, I think both options are feasible, just thought it was worth suggesting for your consideration during your reviews.

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:113]:
```suggestion
   An |MO| is created to produce three units of *wood*, which is used as a component for the *rocking
   chair* product. Clicking the :guilabel:`Allocation` smart button on the |MO| opens an allocation
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:114]:
```suggestion
   report that lists open *rocking chair* |MOs| that require one or more pieces of *wood*.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:83]:
```suggestion
   delivery orders that require one or more *rocking chairs*.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:135]:
```suggestion
After clicking :guilabel:`Assign All` or :guilabel:`Assign`, the :guilabel:`Print Labels` (or
:guilabel:`Print Label`) button becomes selectable. Selecting either button
generates and downloads a PDF document, with one label for each product that was assigned. These
```

**Review by ksc-odoo — APPROVED** [review]
hey @hojo-odoo -- just finished my Final Review. Nice work. Very detailed. Approving now. Just have a small collection of feedback/comments that require your attention. But, once you implement the necessary changes, you can feel free to tag this for Tech Review. Thanks! 👍

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (hojo-odoo) [reply content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:32]: It can be, but I prefer to always have inventory actions conducted through the Inventory app.

**Comment by hojo-odoo** [comment]
Hey @samueljlieber,

This doc is ready for technical review.

Thanks,
John

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:55]: I understand this is important to distinguish, but I think the all-cap is a bit much. Suggesting to just leverage the bold formatting for emphasis :) 
```suggestion
If there **are** existing |SOs| and |MOs| that require the product, **and** there are too few units
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:62]:
```suggestion
If there **are** existing |SOs| and |MOs| that require the product, **and** there are enough units
of the product on-hand to fulfill those orders, then the :icon:`fa-list` :guilabel:`Allocation`
smart button only appears at the top of the page once the |MO| has been marked as done, by clicking
:guilabel:`Produce All`.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:69]:
```suggestion
   If there **are not** any existing |SOs| and |MOs| that require the product, the :icon:`fa-list`
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:30]: I dont think this image adds much value, since the instruction is very clear. Can remove :)
```suggestion
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/workflows/allocation.rst:39]: Same with this one, can remove :)
```suggestion
```

**Review by samueljlieber — APPROVED** [review]
Hi @hojo-odoo! Awesome work on this new Manufacturing doc on Allocation reports. Your PR looks great to me! Approving with just a couple nitpicks on formatting and images, thank you for your work!
.....
@robodoo delegate=hojo-odoo

**Comment by hojo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
